proposals/0012-introduce-reactNativeMetadata.md

```diff
@@ -0,0 +1,403 @@
+
+## Proposed Tooling
+
+This section lists the tooling we believe could be built on top of `reactNativeManifest` to improve the developer experience.
+
+### Build Logic support
+
+Build tools such as Gradle/CocoaPods or others should account for the `reactNativeManifest` section and use it to configure the build pipeline accordingly to this and future RFCs. Specifically they should follow this logic for a generic capability `foo`.
+
+1. If the capability `foo` is specificed in the `reactNativeManifest.capabilities` section, use the value specified in the `reactNativeManifest.capabilities.foo` section.
+   1. If the user is **also** specifying the capability `foo` in the build tool specific configuration (e.g. inside `gradle.properties` for Android) behave as follows:
+      1. If the two values are **compatible**, use them without notifying the user.
+      1. If the two values are **incompatible**, notify the user and use the value specified in the `reactNativeManifest.capabilities.foo` section will prevail and the value specified in the build tool specific configuration will be ignored.
+1. If the capability `foo` is **not** specificed in the `reactNativeManifest.capabilities` section, honor the value specified in the build tool specific configuration or the default value if not specified.
+
+### `align-deps` support
+
+This RFC originated from a [conversation between Meta and Microsoft](https://github.com/microsoft/rnx-kit/issues/1863) to use `align-deps` as key feature in the New Architecture rollout support, in helping developers understand if a library they're using is compatible with the New Architecture or not. Very quickly it became clear that there is currently no straightforward way to know if a library indeed has that support.
+
+Therefore we believe `align-deps` can benefit from this information and can be extended to offer New Architecture support information.
+
+### React Native CLI support
+
+Ideally the CLI (or any other build tool) should be able to consume the `reactNativeManifest` section and warn at build time if a configuration is invalid (either by having an invalid value or by having a mixture of invalid configurations).
+For example, the CLI could use [jsonschema](https://npmjs.com/package/jsonschema) at build time and validate the `package.json` against the schema we provide.
+
+It's outside the scope of this RFC to delve into details of which tool will implement which logic and we defer this to a future discussion.
+
+### React Native Directory
+
+The React Native direcotry can easily consume the `reactNativeManifest` by:
+1. Query the NPM registry with [their API](https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md) to obtain the repository information.
+2. Query the repository information to obtain the `packge.json` and read the `reactNativeManifest` section.
+
```

Comment:
maybe worth detailing that it's precisely because it won't require to download the whole package/node_module that we are putting this config in the package.json